config/v1/types_cluster_operator.go

```diff
@@ -176,10 +176,16 @@ const (
 	// unexpected errors are handled as operators mature.
 	OperatorDegraded ClusterStatusConditionType = "Degraded"
 
-	// Upgradeable indicates whether the operator safe to upgrade based on the current cluster state. When status is `False`
-	// administrators should not upgrade their cluster and the message field should contain a human readable description
-	// of what the administrator should do to allow the operator to successfully update.  A missing condition, True,
-	// and Unknown are all treated by the CVO as allowing an upgrade.
+	// Upgradeable indicates whether the operator is safe to upgrade based on the
+	// current cluster state. When status is False, the cluster-version operator
+	// will prevent the cluster from performing impacted updates unless forced.
+	// When set on ClusterVersion, the message will explain which updates (minor
+	// or patch) are impacted. When set on ClusterOperator, False will block
```

Comment:
Is there an implicit hierarchy here that needs to be called out? For example, if minor version updates are not allowed, are patch version updates also automatically disallowed ?

Reply:
> For example, if minor version updates are not allowed, are patch version updates also automatically disallowed ?

[SemVer is `MAJOR.MINOR.PATCH`][1], so the two cases we have today are "blocks minor bumps, patch bumps are fine" and "blocks all bumps".  So could have been `minor or all` here.  The `.spec.overrides`-is-set message [is][2]:

> Disabling ownership via cluster version overrides prevents upgrades. Please remove overrides before continuing.

So the actual user experience should be unambiguous.

[1]: https://semver.org/spec/v2.0.0.html#summary
[2]: https://github.com/openshift/cluster-version-operator/blob/513a2fc5e2e8f4e8fad5c30913d8129c110b0684/pkg/cvo/upgradeable.go#L232